diff --git a/elf/elf.h b/elf/elf.h
index 89fc8021e9..79458701e3 100644
--- a/elf/elf.h
+++ b/elf/elf.h
@@ -4205,6 +4198,15 @@ enum
 #define R_LARCH_TLS_GD_HI20 98
 #define R_LARCH_32_PCREL 99
 #define R_LARCH_RELAX 100
+#define R_LARCH_DELETE      101
+#define R_LARCH_ALIGN       102
+#define R_LARCH_PCREL20_S2  103
+#define R_LARCH_CFA         104
+#define R_LARCH_ADD6        105
+#define R_LARCH_SUB6        106
+#define R_LARCH_ADD_ULEB128 107
+#define R_LARCH_SUB_ULEB128 108
+#define R_LARCH_64_PCREL    109
 
 /* ARC specific declarations.  */
 
