Index: pacman-6.0.1/lib/libalpm/package.c
===================================================================
--- pacman-6.0.1.orig/lib/libalpm/package.c
+++ pacman-6.0.1/lib/libalpm/package.c
@@ -71,6 +71,7 @@ int SYMEXPORT alpm_pkg_checkmd5sum(alpm_
 	return retval;
 }
 
+static char loongarch[] = "loong64";
 /* Default package accessor functions. These will get overridden by any
  * backend logic that needs lazy access, such as the local database through
  * a lazy-load cache. However, the defaults will work just fine for fully-
@@ -81,7 +82,13 @@ static const char *_pkg_get_url(alpm_pkg
 static alpm_time_t _pkg_get_builddate(alpm_pkg_t *pkg)   { return pkg->builddate; }
 static alpm_time_t _pkg_get_installdate(alpm_pkg_t *pkg) { return pkg->installdate; }
 static const char *_pkg_get_packager(alpm_pkg_t *pkg)    { return pkg->packager; }
-static const char *_pkg_get_arch(alpm_pkg_t *pkg)        { return pkg->arch; }
+static const char *_pkg_get_arch(alpm_pkg_t *pkg)        {
+	if (strcmp(pkg->arch, "loongarch64") == 0) {
+		return loongarch;
+	} else {
+		return pkg->arch;
+	}
+}
 static off_t _pkg_get_isize(alpm_pkg_t *pkg)             { return pkg->isize; }
 static alpm_pkgreason_t _pkg_get_reason(alpm_pkg_t *pkg) { return pkg->reason; }
 static int _pkg_get_validation(alpm_pkg_t *pkg) { return pkg->validation; }
Index: pacman-6.0.1/src/pacman/conf.c
===================================================================
--- pacman-6.0.1.orig/src/pacman/conf.c
+++ pacman-6.0.1/src/pacman/conf.c
@@ -993,6 +1003,11 @@ static int _parse_repo(const char *key,
 	if(strcmp(key, "Server") == 0) {
 		CHECK_VALUE(value);
 		repo->servers = alpm_list_add(repo->servers, strdup(value));
+		if(strstr(value, "$arch") != NULL) {
+			char *replaced;
+			replaced = strreplace(value, "$arch", "loongarch64");
+			repo->servers = alpm_list_add(repo->servers, replaced);
+		}
 	} else if(strcmp(key, "SigLevel") == 0) {
 		CHECK_VALUE(value);
 		alpm_list_t *values = NULL;
