Index: pacman-6.0.1/src/pacman/conf.c
===================================================================
--- pacman-6.0.1.orig/src/pacman/conf.c
+++ pacman-6.0.1/src/pacman/conf.c
@@ -402,7 +402,11 @@ int config_add_architecture(char *arch)
 		struct utsname un;
 		char *newarch;
 		uname(&un);
-		newarch = strdup(un.machine);
+        if (strcmp(un.machine, "loongarch64") == 0) {
+            newarch = strdup("loong64");
+        } else {
+            newarch = strdup(un.machine);
+        }
 		free(arch);
 		arch = newarch;
 	}
