From 1566f727930285dcfd8185a41e7c75b528158c4c Mon Sep 17 00:00:00 2001
From: Zhang Na <zhangna@loongson.cn>
Date: Wed, 26 Jan 2022 11:54:19 +0800
Subject: [PATCH] bpo-46498: Add new triplets for loongarch64 Suggested by WANG
 Xuerui

---
 configure    | 14 ++++++++++++++
 configure.ac | 14 ++++++++++++++
 2 files changed, 28 insertions(+)

diff --git a/configure b/configure
index 78e5a09927221..01c538dce66e3 100755
--- a/configure
+++ b/configure
@@ -5993,6 +5993,20 @@ cat >> conftest.c <<EOF
         hppa-linux-gnu
 # elif defined(__ia64__)
         ia64-linux-gnu
+# elif defined(__loongarch__)
+#  if defined(__loongarch_lp64)
+#   if defined(__loongarch_soft_float)
+        loongarch64-linux-gnusf
+#   elif defined(__loongarch_single_float)
+        loongarch64-linux-gnuf32
+#   elif defined(__loongarch_double_float)
+        loongarch64-linux-gnu
+#   else
+#    error unknown platform triplet
+#   endif
+#  else
+#   error unknown platform triplet
+#  endif
 # elif defined(__m68k__) && !defined(__mcoldfire__)
         m68k-linux-gnu
 # elif defined(__mips_hard_float) && defined(__mips_isa_rev) && (__mips_isa_rev >=6) && defined(_MIPSEL)
diff --git a/configure.ac b/configure.ac
index a0fbe41c9ec59..c4fe61bef2d91 100644
--- a/configure.ac
+++ b/configure.ac
@@ -884,6 +884,20 @@ cat >> conftest.c <<EOF
         hppa-linux-gnu
 # elif defined(__ia64__)
         ia64-linux-gnu
+# elif defined(__loongarch__)
+#  if defined(__loongarch_lp64)
+#   if defined(__loongarch_soft_float)
+        loongarch64-linux-gnusf
+#   elif defined(__loongarch_single_float)
+        loongarch64-linux-gnuf32
+#   elif defined(__loongarch_double_float)
+        loongarch64-linux-gnu
+#   else
+#    error unknown platform triplet
+#   endif
+#  else
+#   error unknown platform triplet
+#  endif
 # elif defined(__m68k__) && !defined(__mcoldfire__)
         m68k-linux-gnu
 # elif defined(__mips_hard_float) && defined(__mips_isa_rev) && (__mips_isa_rev >=6) && defined(_MIPSEL)
Index: Python-3.10.2/Modules/_ctypes/callproc.c
===================================================================
--- Python-3.10.2.orig/Modules/_ctypes/callproc.c
+++ Python-3.10.2/Modules/_ctypes/callproc.c
@@ -1112,7 +1112,7 @@ GetComError(HRESULT errcode, GUID *riid,
 #endif
 
 #if (defined(__x86_64__) && (defined(__MINGW64__) || defined(__CYGWIN__))) || \
-    defined(__aarch64__) || defined(__riscv)
+    defined(__aarch64__) || defined(__riscv) || defined(__loongarch64)
 #define CTYPES_PASS_BY_REF_HACK
 #define POW2(x) (((x & ~(x - 1)) == x) ? x : 0)
 #define IS_PASS_BY_REF(x) (x > 8 || !POW2(x))
