Index: glib/gio/tests/meson.build
===================================================================
--- glib.orig/gio/tests/meson.build
+++ glib/gio/tests/meson.build
@@ -736,6 +736,7 @@ if not meson.is_cross_build() or meson.h
       output : 'test_resources2.o',
       command : [objcopy,
                  '--add-symbol','_g_binary_test1_resource_data=.data:0',
+                 '--alt-elf-eflags=0x03',
                  '@INPUT@',
                  '@OUTPUT@'])
 
