Index: dhcpcd-9.4.1/src/privsep-linux.c
===================================================================
--- dhcpcd-9.4.1.orig/src/privsep-linux.c
+++ dhcpcd-9.4.1/src/privsep-linux.c
@@ -188,6 +188,8 @@ ps_root_sendnetlink(struct dhcpcd_ctx *c
 #  define SECCOMP_AUDIT_ARCH AUDIT_ARCH_IA64
 #elif defined(__microblaze__)
 #  define SECCOMP_AUDIT_ARCH AUDIT_ARCH_MICROBLAZE
+#elif defined(__loongarch64)
+#  define SECCOMP_AUDIT_ARCH AUDIT_ARCH_LOONGARCH64
 #elif defined(__m68k__)
 #  define SECCOMP_AUDIT_ARCH AUDIT_ARCH_M68K
 #elif defined(__mips__)
